src/dnf-plugins/subscription-manager.py

```diff
@@ -142,14 +142,21 @@ def _warnOrGiveUsageMessage(self):
             if msg:
                 logger.info(msg)
 
+    def sack(self):
+        # If command demands resolving, it should run transaction. This cannot
+        # be run in a Plugin.config hook as currently the Plugin.config is
+        # called before Command.configure.
+        if not self.cli.demands.resolving:
+            self._package_profile()
```

Comment:
We should make sure not to run this twice. I'd suggest tracking if we've run this already as an attribute of the plugin class itself which is checked in the _package_profile method.

Reply:
This is not a problem as command either has transaction or it doesn't. The cases are mutually exclusive. If there is no transaction then profile is uploaded in sack phase. If there is a transaction the profile is uploaded after the transaction ends.